src/status_im/contexts/wallet/send/input_amount/view.cljs

```diff
@@ -248,11 +303,22 @@
            :left-action          :dot
            :delete-key?          true
            :on-press             (fn [c]
-                                   (when-not loading-routes?
-                                     (set-input-state #(controlled-input/add-character % c))))
+                                   (let [new-text      (str input-amount c)
+                                         max-decimals  (if @crypto-currency? crypto-decimals 2)
+                                         regex-pattern (str "^\\d*\\.?\\d{0," max-decimals "}$")
```

Comment:
When typing, we limit the input to allow decimals up to the decimals amount of the token or 2 if it is a fiat amount.